pkg/network/ebpf/c/protocols/protocol-classification-helpers.h

```diff
@@ -88,6 +93,53 @@ static __always_inline bool is_http(const char *buf, __u32 size) {
     return http;
 }
 
+// Regular format of postgres message: | byte tag | int32_t len | string payload |
+// From https://www.postgresql.org/docs/current/protocol-overview.html:
+// The first byte of a message identifies the message type, and the next four bytes specify the length of the rest
+// of the message (this length count includes itself, but not the message-type byte). The remaining contents of the
+// message are determined by the message type
+struct pg_message_header {
+    char message_tag;
+    uint32_t message_len; // Big-endian: use bpf_ntohl to read this field
+    char payload[0];
+} __packed;
+
+static __always_inline bool is_postgres(const char *buf, __u32 buf_size) {
+    CHECK_PRELIMINARY_BUFFER_CONDITIONS(buf, buf_size, POSTGRES_MIN_MSG_SIZE);
+
+    const struct pg_message_header *hdr = (struct pg_message_header *)buf;
+
+    switch (hdr->message_tag) {
+    case POSTGRES_BIND_MAGIC_BYTE:
+    case POSTGRES_CLOSE_MAGIC_BYTE:
+    case POSTGRES_DESCRIBE_MAGIC_BYTE:
+    case POSTGRES_EXECUTE_MAGIC_BYTE:
+    case POSTGRES_COPY_FAIL_MAGIC_BYTE:
+    case POSTGRES_FLUSH_MAGIC_BYTE:
+    case POSTGRES_PARSE_MAGIC_BYTE:
+    case POSTGRES_PASSWORD_MESSAGE_MAGIC_BYTE:
+    case POSTGRES_QUERY_MAGIC_BYTE:
+    case POSTGRES_SYNC_MAGIC_BYTE:
+    case POSTGRES_TERMINATE_MAGIC_BYTE:
+        break;
+    default:
+        return false;
+    }
+
+    uint32_t message_len = bpf_ntohl(hdr->message_len);
+    if (message_len < POSTGRES_MIN_PAYLOAD_LEN || message_len > POSTGRES_MAX_PAYLOAD_LEN) {
+        return false;
+    }
+
+    char message_last_byte = buf[message_len];
+    // If the input includes a whole message (1 byte tag + length), check the last character.
+    if ((message_len + 1 <= (int)buf_size) && (message_last_byte != '\0')) {
+        return false;
+    }
+
+    return true;
```

Comment:
So that was true for Seekret code (use syscall)
but now we are using socket filter, and large payloads will probably be splited into multiple tcp packets
we should find a better check

I suggest to read the protocol documentation and understand if there are small messages (by design) that are always being sent